daffodil-propgen/src/main/scala/org/apache/daffodil/propGen/TunableGenerator.scala

```diff
@@ -134,7 +138,9 @@ class TunableGenerator(schemaRootConfig: scala.xml.Node, schemaRootExt: scala.xm
     val tunables =
       tunableNodes.map { tunableNode =>
         val schemaName = tunableNode \@ "name"
-        val schemaType = tunableNode \@ "type"
+        val schemaType =
+          if (tunableNode \@ "type" != "") tunableNode \@ "type"
+          else tunableNode \\ "restriction" \@ "base"
```

Comment:
This code is not going to give any sort of useful diagnostic if the XSD just happens to have constructs that aren't recognized in it. 

Add a check here if there is no type attribute, then there must be a restriction, or throw a useful diagnostic stating that these are the only allowed options.